src/main/scala/com/twosigma/flint/rdd/RangeDependency.scala

```diff
@@ -39,19 +38,9 @@ private[rdd] object RangeDependency {
     normalizationStrategy: PartitionNormalizationStrategy = HeavyKeysNormalizationStrategy
   )(implicit ord: Ordering[K]): Seq[RangeDependency[K, P]] = {
     require(headers.nonEmpty, "Need at least one partition")
+    import OrderedPartitionHeaderUtils.HeaderOrdering
 
-    val sortedHeaders = headers.sortBy(_.partition.index).toArray
-    // Assume partitions are sorted, i.e. the keys of ith partition are less or equal than those of (i + 1)th partition.
-    sortedHeaders.reduceOption {
-      (h1, h2) =>
-        if (ord.lteq(h1.firstKey, h2.firstKey)) {
-          h2
-        } else {
-          sys.error(s"Partitions are not sorted. " +
-            s"The partition ${h1.partition.index} has the first key ${h1.firstKey} and " +
-            s"the partition ${h2.partition.index} has the first key ${h2.firstKey}.")
-        }
-    }
+    val sortedHeaders = headers.sortBy(x => x).toArray
```

Comment:
I think we're ok, so we start with
`headers = Header(Partition(0), 100, ...)), Header(Partition(1), 0, ...)`

`sortedHeaders = Header(Partition(1), 0, ...), Header(Partition(0), 100, ...)`

Further down we get:
`val normalizedRanges = normalisationStrategy.normalise(sortedHeaders)
= [0, 100) [100, 200)`
None of the normalisation strategies look at the partition index following my refactor

We then return our RangeDependencies from this method:

`RangeDependency[K, P](idx, normalizedRange, parents)`

Which gives us
`RangeDepdency(0, [0, 100), (Partition(1))), RangeDependency(1, [100, 200), (Partition(0))

Where idx is what we've got from zipWithIndex, so at this point we have re-index things to have the correctly ordered indexes, and the rest of the program can continue happily. And indeed further up the stack with in Conversion.scala we build our RangeSplit off the index of the RangeDependency:
`d => RangeSplit(OrderedRDDPartition(d.index).asInstanceOf[Partition], d.range)`

Which would give us
RangeSplit(OrderedRDDPartion(0), [0, 100)) RangeSplit(OrderedRDDPartion(1), [100, 200))

Just as we want.


Or is your claim that we need to have the partitions remapped before we pass them into the normalisation strategy? I feel like the normalisation strategies should just rely on data from the headers and not info about the partitions themselves.